src/Flex.php

```diff
@@ -653,6 +662,36 @@ public function generateFlexId()
         $this->updateComposerLock();
     }
 
+    private function updateAutoloadFile()
```

Comment:
shouldn't this be hooked into `POST_AUTOLOAD_DUMP` instead so that it is still applied when using `dump-autoload` ?